docs(applying-storybook): update storybook docs for v7

TylerAPfledderer opened the pull request:
_This is an update for #9546 to provide better direction in implementation for Storybook for components._

This PR updates the [applying-storybook.md](https://github.com/ethereum/ethereum-org-website/blob/dev/docs/applying-storybook.md) in the docs directory of the project to be inline with Storybook v7

- Updates codeblocks and notation for the new Story Object syntax
- Adds a quick ref table for the different selections in the UI Dashboard

TylerAPfledderer commented:
@pettinarip I would love to know if there is anything I should add or remove. :)

TylerAPfledderer commented:
@pettinarip I have made those requested changes.

One final thing I suppose that is missing. Should I add on here notation regarding what the directory structure should be in Storybook per Nuno's suggestions, and include how to write them in each stories meta `title` option?

It's currently only set up to do an auto generate based on the current structure which will not have the intended result.

Disclaimer: I think there is a way we can auto-generate the correct structure too, but might require known variables to exist. And I haven't explored this kind of feature before.

pettinarip commented:
@TylerAPfledderer that is a good point.

I don't think we need to define the concrete structure here but perhaps it is worth to mention that our naming convention is coming from the DS. That we follow the naming declared in the DS.

That way, we avoid the maintenance of this doc.

TylerAPfledderer commented:
I'll ping you when I've added a blurb for this.

TylerAPfledderer commented:
@pettinarip I've added the note for the display name preference.

I've also added a note regarding [single-story hoisting of the display name](https://github.com/ethereum/ethereum-org-website/pull/9575/files#diff-be6f8c53bd1da59e25e89779449738be4716663b7c000e38745c7534e435e2d3R109).

pettinarip reviewed (approved):
Awesome! great job, thanks @TylerAPfledderer
